mne/tests/test_cov.py

```diff
@@ -527,15 +547,16 @@ def test_compute_covariance_auto_reg(rank):
 
 
 def _cov_rank(cov, info):
-    # XXX : this should use the to appear compute_rank function
-    return compute_whitener(cov, info, return_rank=True, verbose='error')[2]
+    with pytest.warns(None):  # ignore warnings about rank mismatches
```

Comment:
why do we have the warnings? Why not calling directly the test `_compute_rank_int(cov, info=info)`?

Reply:
Because when using some modes it warns when computing the rank (and we are intentionally ignoring it)